projects/extension/src/containers/Popup.tsx

```diff
@@ -24,91 +89,114 @@ const Popup: FunctionComponent = () => {
 
       chrome.runtime.getBackgroundPage((backgroundPage) => {
         if (!isActive) return
-
         const bg = backgroundPage as Background
-        disconnectTabRef.current = bg.uiInterface.disconnectTab
-
-        const refresh = () => {
-          const networksByTab: Map<number, Set<string>> = new Map()
-          bg.uiInterface.chains.forEach((app) => {
-            if (!app.tab) return
-            if (!networksByTab.has(app.tab.id))
-              networksByTab.set(app.tab.id, new Set())
-            networksByTab.get(app.tab.id)!.add(app.chainName)
-          })
-
-          const nextTabs: TabInterface[] = []
-          browserTabs.forEach((tab) => {
-            if (!networksByTab.has(tab.id!)) return
-            const result = {
-              tabId: tab.id!,
-              url: tab.url,
-              networks: [...networksByTab.get(tab.id!)!],
-            }
-            if (tab.active) setActiveTab(result)
-            else nextTabs.push(result)
-          })
-
-          setOtherTabs(nextTabs)
-        }
+        disconnectTab.current = bg.uiInterface.disconnectTab
         unsubscribe = bg.uiInterface.onChainsChanged(refresh)
         refresh()
       })
     })()
 
     return () => {
       isActive = false
-      unsubscribe()
+      unsubscribe && unsubscribe()
     }
   }, [])
 
   const goToOptions = (): void => {
     chrome.runtime.openOptionsPage()
   }
 
+  const networkIcon = (network: string) => {
+    const icon = network.replace(new RegExp(" ", "g"), "").toLowerCase()
+    return (
+      <>
+        <div className="icon w-7">
+          {knownChains.includes(icon) ? icon : "?"}
+        </div>
+        <div className="pl-2">{network}</div>
```

Comment:
I'm not super confident in this. This value is untrusted and entirely decided by the website. A website could have a name that is super long or has tons of weird unicode characters.

Reply:
Im not sure I understand - this part of code is showing the network name. So even if there is a custom chain/parachain that is named: `123AS$@#$SomethingMoreAnd123;'_'ThisCanGoOnUntilItIsVeryLong`, still will just appear as the "network name" and icon as `?`. I'm not sure what the vulnerability is

Reply:
There's no "vulnerability", just someone could completely destroy your CSS with a very long name, unless you made sure that it can't happen